c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -954,11 +954,63 @@ bool SetBackpressure(JSContext *cx, HandleObject stream, bool backpressure);
 bool Error(JSContext *cx, HandleObject stream, HandleValue error);
 } // namespace TransformStream
 
+#define ITERTYPE_ENTRIES 0
+#define ITERTYPE_KEYS 1
+#define ITERTYPE_VALUES 2
 namespace URLSearchParams {
-bool is_instance(JSObject *obj);
-JSObject *create(JSContext *cx, jsurl::JSUrl *url);
+
+namespace Slots {
+enum { Url, Params, Count };
+};
+
 JSUrlSearchParams *get_params(JSObject *self);
+
+namespace detail {
+bool append(JSContext *cx, HandleObject self, HandleValue key, HandleValue val, const char *_);
+} // namespace detail
+
 SpecSlice serialize(JSContext *cx, HandleObject self);
+
+bool check_receiver(JSContext *cx, HandleValue receiver, const char *method_name);
+
+const unsigned ctor_length = 1;
+
+bool append(JSContext *cx, unsigned argc, Value *vp);
+
```

Comment:
I was wondering where the implementations for these prototypes were, and it's down on line 6935 on main -- the `URLSearchParams` namespace is reopened three times throughout `js-compute-builtins.cpp`. We should do some reorganization soon so that we don't need prototypes so far from the definitions :)